ui/selectors/selectors.js

```diff
@@ -1278,13 +1279,59 @@ export function getAllAccountsOnNetworkAreEmpty(state) {
   return hasNoNativeFundsOnAnyAccounts && hasNoTokens;
 }
 
-export function getShouldShowSeedPhraseReminder(state) {
-  const { tokens, seedPhraseBackedUp, dismissSeedBackUpReminder } =
-    state.metamask;
+export function getIsAnyAccountWithBalanceOnNonTestNetwork(state) {
+  const { cachedBalances } = state.metamask;
+
+  const zeroBalanceCheck = (currentValue) =>
+    currentValue === '0x0' || currentValue === '0x00';
+
+  const balancesOnNonTestNetworks = Object.entries(cachedBalances)
+    .filter(
+      (networkAndAccountBalances) =>
+        TEST_CHAINS.includes(networkAndAccountBalances[0]) === false &&
+        networkAndAccountBalances,
+    )
+    .map((accountAndBalance) => Object.values(accountAndBalance[1]))
+    .map((balance) => balance.every(zeroBalanceCheck));
+
+  return balancesOnNonTestNetworks.includes(false);
+}
+
+export async function getIsAnyTokenWithNonZeroBalance(state) {
+  const { tokens } = state.metamask;
+  const selectedAddress = getSelectedAddress(state);
+
+  if (tokens.length > 0) {
```

Comment:
Is this `if` required if the `length` is `0`?

Reply:
No, because we just need to know if there is some token with non-zero balance. If the length of the token array is 0, then the return is false, as expected.

Reply:
I agree with @darkwing here, the `if` is not necessary.
If `tokens.length` is 0, then the loop evaluates to
`for (let i = 0; i < 0; i++)`
which is a loop that runs 0 times.